File: mobile/mobile/Models/Event.swift
import Foundation

struct Event: Codable, Identifiable, Hashable {
    let id: Int?
    let name: String
    let nameAr: String?
    let description: String?
    let descriptionAr: String?
    let code: String
    let poster: String?
    let date: String?
    let published: Bool
    let locationId: Int?
    let location: Location?
    let speakers: [EventSpeaker]?
    let isActive: Bool?
    let registrationCount: Int?
    let createdAt: String?
}

struct Location: Codable, Hashable {
    let id: Int?
    let name: String?
    let nameAr: String?
}

struct EventSpeaker: Codable, Hashable {
    let id: Int?
    let name: String
    let nameAr: String?
}

File: backend/UMS/Models/Event.cs
using System.Text.Json.Serialization;
using UMS.Models.Shared;

namespace UMS.Models;

public class Event : BaseModel
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string? NameAr { get; set; }
    public string? Description { get; set; }
    public string? DescriptionAr { get; set; }
    public string Code { get; set; } // Unique code for public access
    public string? Poster { get; set; } // Path to poster file
    public string? Badge { get; set; } // Path to badge template file
    public string? Agenda { get; set; } // Path to agenda PDF file
    public DateTime? Date { get; set; } // Event date
    public bool Published { get; set; } = false;
    public int? LocationId { get; set; }
    public Location? Location { get; set; }

    [JsonIgnore]
    public ICollection<EventSpeaker> Speakers { get; set; } = new List<EventSpeaker>();

    [JsonIgnore]
    public ICollection<EventRegistration> Registrations { get; set; } = new List<EventRegistration>();

    [JsonIgnore]
    public ICollection<EventSession> Sessions { get; set; } = new List<EventSession>();
}

